### PR TITLE
Update: Adding compatibility for Angular 8

### PR DESCRIPTION
**Pull request by zackarychapple**
**What kind of change does this PR introduce?** (check one with "x")
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
The current version is pulling DOCUMENT from platform-browser. With angular 8 this is deprecated and now pulls from common. This PR corrects for the new import. 

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x")
- [x] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...
Requires Angular >8

**Other information**:

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -66,23 +66,17 @@
...
   "devDependencies": {
-    "@angular/animations": "^7.0.0",
-    "@angular/common": "^7.0.0",
-    "@angular/compiler": "^7.0.0",
-    "@angular/compiler-cli": "^7.0.0",
-    "@angular/core": "^7.0.0",
-    "@angular/platform-browser": "^7.0.0",
-    "@angular/platform-browser-dynamic": "^7.0.0",
-    "@angular/platform-server": "^7.0.0",
+    "@angular/animations": "^8.0.0-rc.3",
```
Comment (whyboris) [inline package.json]: Angular 8 just came out -- perhaps bump these to `"^8.0.0" ? 🙂

**Comment by whyboris** [comment]
🙇 Dear *swimlane* team -- our production project can not upgrade to _Angular 8_ because this PR hasn't been merged. Please merge it soon and draft a new _npm_ release 🙇 

❤️ much appreciated ❤️

**Comment by marjan-georgiev** [comment]
I am currently working on this. Angular 8 seems to require a few more changes (adding the `static` option to contentchild and viewchild directives). Should have it ready soon.

**Comment by marjan-georgiev** [comment]
See https://github.com/swimlane/ngx-datatable/pull/1697

Would appreciate if anyone would give it a test before I merge and release.

**Pull request closed without merging**